eemeter/modeling/models/billing_caltrack.py

```diff
@@ -0,0 +1,308 @@
+        ''' Fits a model to the input data.
+
+        Parameters
+        ----------
+        input_data : pandas.DataFrame
+            Formatted input data as returned by
+            :code:`ModelDataBillingFormatter.create_input()`
+
+        Returns
+        -------
+        out : dict
+            Results of this model fit:
+
+            - :code:`"r2"`: R-squared value from this fit.
+            - :code:`"model_params"`: Fitted parameters.
+
+              - :code:`X_design_matrix`: patsy design matrix used in
+                formatting design matrix.
+              - :code:`formula`: patsy formula used in creating design matrix.
+              - :code:`coefficients`: ElasticNetCV coefficients.
+              - :code:`intercept`: ElasticNetCV intercept.
+
+            - :code:`"rmse"`: Root mean square error
+            - :code:`"cvrmse"`: Normalized root mean square error
+              (Coefficient of variation of root mean square error).
+            - :code:`"upper"`: self.upper,
+            - :code:`"lower"`: self.lower,
+            - :code:`"n"`: self.n
+        '''
+        trace_data, temperature_data = input_data
+
+        data = pd.DataFrame(
+            {'energy': trace_data.iloc[:-1], 'tempF': temperature_data},
+            columns=['energy', 'tempF'])
+        model_data = self._make_monthly(data)
+
+        if np.sum(np.isfinite(model_data['upd'])) <= 12:
+            return None
+
+        # Fit the intercept-only model
+        formula = 'upd ~ 1'
+        try:
+            int_mod = smf.ols(formula=formula, data=model_data)
+            int_res = int_mod.fit()
+            int_rsquared = int_res.rsquared
+            int_qualified = (int_res.params['Intercept'] >= 0) and \
+                            (int_res.pvalues['Intercept'] < 0.1)
+        except:
+            int_rsquared, int_qualified = 0, False
+    
+        # CDD-only
+        formula = 'upd ~ CDD'
+        try:
+            if self.heating_only: assert False
+            cdd_mod = smf.ols(formula=formula, data=model_data)
+            cdd_res = cdd_mod.fit()
+            cdd_rsquared = cdd_res.rsquared
+            cdd_qualified = (cdd_res.params['Intercept'] >= 0) and (cdd_res.params['CDD'] >= 0) and \
+                            (cdd_res.pvalues['Intercept'] < 0.1) and (cdd_res.pvalues['CDD'] < 0.1)
+        except:
+            cdd_rsquared, cdd_qualified = 0, False
+    
+        # HDD-only
+        formula = 'upd ~ HDD'
+        try:
+            hdd_mod = smf.ols(formula=formula, data=model_data)
+            hdd_res = hdd_mod.fit()
+            hdd_rsquared = hdd_res.rsquared
+            hdd_qualified = (hdd_res.params['Intercept'] >= 0) and (hdd_res.params['HDD'] >= 0) and \
+                            (hdd_res.pvalues['Intercept'] < 0.1) and (hdd_res.pvalues['HDD'] < 0.1)
+        except:
+            hdd_rsquared, hdd_qualified = 0, False
+    
+        # CDD+HDD
+        formula = 'upd ~ CDD + HDD'
+        try:
```

Comment:
Where is the error likely to occur in this block? (Just that assert?) That line or those lines should be isolated in the try block and the parts that aren't likely to fail should be placed in an else block.